### PR TITLE
Revising tagging conformance criteria for correctness

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
The original draft of the conformance did not require tagging of files, but somehow it became a requirement on paper. 

Tagging is a really good idea, but it's not actually a requirement of the code on any version of nodeJS because Zowe uses a configuration option by which nodeJS 12+ treats untagged file encoding identical to versions before 12. IBM has stated this option will continue to exist and there's no end in sight for it, so there isn't a reason to force tagging at this time.

Furthermore, there are some technical hurdles around tagging files at runtime. It's pretty easy to tag what you ship, but if files are created at runtime, how they get tagged is implementation-specific and nodejs doesn't have a built-in way to do it, for example.

So for now, this should be marked as a best practice.
If the nodejs flag we use for having consistent untagged behavior ever becomes deprecated, we'll switch this to being a requirement, but there is no date on that happening.

Signed-off-by: 1000TurquoisePogs <sgrady@rocketsoftware.com>

### CODE REVIEW

**Comment by jmertic** [comment]
@1000TurquoisePogs This is something weird with an integration I have for issue management - I can force merge in this to get around this annoyance when @Joe-Winchester and @RASakach comment and approve.

**Review by Joe-Winchester — APPROVED** [review]
I agree with @1000TurquoisePogs that mandating explicit tagging should be **best practice**.   Tagging is difficult, especially for modules that are built off-patform and moving towards containerization of Zowe this will become more frequent for the build pipeline.  As long as node v12 (and future releases) support the backwards compatibility and control then Zowe should make it **best practice**.  I think having an accompanying section in the docs.zowe.org on tagging would be helpful as there are complexities, so I've created https://github.com/zowe/docs-site/issues/1433 for the creation of tagging background docs.  This issue should not hold up this merge and I'm happy to approve this change as-is.

**Comment by jmertic** [comment]
Re-reading this - my only concern is that we are making a "required" into a "best practice", thus changing the requirements for conformance. Generally the accepted changes have been more related to typos and clearing up wording, so this is different. But I defer to @Joe-Winchester and @RASakach if they don't see the concern.

**Comment by Joe-Winchester** [comment]
> Re-reading this - my only concern is that we are making a "required" into a "best practice", thus changing the requirements for conformance. Generally the accepted changes have been more related to typos and clearing up wording, so this is different. But I defer to @Joe-Winchester and @RASakach if they don't see the concern.

@jmertic - I fully understand and share your concerns.  I'm working on the TSC charter at the moment and as principle I think it should include voting on changes like this that do affect whether someone who was previously refused conformance and turned away, could be concerned that someone later achieved conformance because the criteria had been relaxed.  
Have you received any submissions from any App-Framework Zowe v1 offerings where conformance was refused ?  Given that the program is recent, that the App-Framework currently has 5 offerings (across four companies - it's the most vendor diverse conformant component)
Given that node v12 that Sean refers to was created post conformance creation and it's the node v12 feature that allows relaxing of conformance (there were discussion between the Zowe team and the Node team for them to implement this feature in node v12 and for them to back up that it'll be supported)  this change can go through as-is, but in future we should defer to the TSC for a wider discussion so it's +1 approve from me for this PR as-is.

**Comment by jmertic** [comment]
No worries @Joe-Winchester - Just want to make everyone away of that point before merging. Will do that now.

**Pull request merged**
